### PR TITLE
Configurable search timeout (#813)

### PR DESCRIPTION
**Pull request by farshidz**
Applying configurable search timeout feature to 2.2. See [original PR](https://github.com/marqo-ai/marqo/pull/813)

### CODE REVIEW

File: src/marqo/core/structured_vespa_index/structured_vespa_index.py [hunk 1]
```diff
@@ -295,7 +295,7 @@ def _to_vespa_tensor_query(self, marqo_query: MarqoTensorQuery) -> Dict[str, Any
 
         if not marqo_query.approximate:
             query['ranking.softtimeout.enable'] = False
-            query['timeout'] = '300s'
+            query['timeout'] = 300 * 1000  # 5 minutes
```
Comment (wanliAlex) [inline src/marqo/core/structured_vespa_index/structured_vespa_index.py:298]: FYI you change the value type from `str` to `int`. If the test can pass, then it's fine.

File: src/marqo/vespa/vespa_client.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def __init__(self, current_generation: int, wanted_generation: int, converged: b
             self.converged = converged
 
     def __init__(self, config_url: str, document_url: str, query_url: str,
-                 content_cluster_name: str, pool_size: int = 10):
+                 content_cluster_name: str, default_search_timeout_ms: int = 1000, pool_size: int = 10):
```
Comment (wanliAlex) [inline src/marqo/vespa/vespa_client.py:40]: doc string args is missing

File: src/marqo/vespa/vespa_client.py [hunk 3]
```diff
@@ -142,7 +143,7 @@ def wait_for_application_convergence(self, timeout: int = 120) -> None:
         raise VespaError(f"Vespa application did not converge within {timeout} seconds")
 
     def query(self, yql: str, hits: int = 10, ranking: str = None, model_restrict: str = None,
-              query_features: Dict[str, Any] = None, **kwargs) -> QueryResult:
+              query_features: Dict[str, Any] = None, timeout: float = None, **kwargs) -> QueryResult:
```
Comment (wanliAlex) [inline src/marqo/vespa/vespa_client.py]: this timeout is `float` while others are `int`

**Review by wanliAlex — CHANGES_REQUESTED** [review]

**Review by pandu-k — APPROVED** [review]

File: src/marqo/core/structured_vespa_index/structured_vespa_index.py (hunk 1 above)
Comment (farshidz) [reply src/marqo/core/structured_vespa_index/structured_vespa_index.py:298]: The client is making a string with `ms` suffix, that's why

File: src/marqo/vespa/vespa_client.py (hunk 2 above)
Comment (farshidz) [reply src/marqo/vespa/vespa_client.py:40]: Need to address this in mainline

**Review by wanliAlex — APPROVED** [review]

File: src/marqo/vespa/vespa_client.py (hunk 3 above)
Comment (farshidz) [reply src/marqo/vespa/vespa_client.py]: Addressed

**Pull request merged**